device/frame/Frame.cpp

```diff
@@ -221,47 +221,55 @@ void Frame::renderFrame()
       camera.Print();
 #endif
 
-      for (const auto& instance : instances)
-      {
-        if (instance->group() == nullptr)
+      vtkm::rendering::Scene surface_scene;
+      vtkm::rendering::Scene volume_scene;
+
+      for (const auto &instance : instances) {
+        if (!instance->isValid()) {
+          reportMessage(ANARI_SEVERITY_DEBUG, "skip rendering invalid group");
           continue;
+        }
 
-        for (const auto& volume : instance->group()->volumes())
-        {
-          const auto actor = volume->actor();
-          const auto mapper = volume->mapper();
-          vtkm::rendering::Scene scene;
-          scene.AddActor(*actor);
-
-          vtkm::rendering::View3D view(scene,
-              *mapper,
-              this->Canvas,
-              camera,
-              this->m_renderer->background());
-          view.SetWorldAnnotationsEnabled(false);
-          view.SetRenderAnnotationsEnabled(false);
-          view.Paint();
+        for (const auto &surface : instance->group()->surfaces()) {
+          if (!surface || !surface->isValid()) {
+            reportMessage(
+                ANARI_SEVERITY_DEBUG, "skip rendering invalid surface");
+            continue;
+          }
+          surface_scene.AddActor(*surface->geometry()->actor());
         }
 
-        for (const auto& surface : instance->group()->surfaces())
-        {
-          std::cout<<"Render Surface"<<std::endl;
-          const auto geom = surface->geometry();
-          const auto actor = geom->actor();
-          const auto mapper = geom->mapper();
-          vtkm::rendering::Scene scene;
-          scene.AddActor(*actor);
-
-          vtkm::rendering::View3D view(scene,
-              *mapper,
-              this->Canvas,
-              camera,
-              this->m_renderer->background());
-          view.SetWorldAnnotationsEnabled(false);
-          view.SetRenderAnnotationsEnabled(false);
-          view.Paint();
+        for (const auto &volume : instance->group()->volumes()) {
+          if (!volume || !volume->isValid()) {
+            reportMessage(
+                ANARI_SEVERITY_DEBUG, "skip rendering invalid volume");
+            continue;
+          }
+          volume_scene.AddActor(*volume->actor());
         }
       }
+
+      if (surface_scene.GetNumberOfActors() != 0) {
+        vtkm::rendering::View3D surface_view(surface_scene,
+            vtkm::rendering::MapperRayTracer(),
```

Comment:
This is fine for now, but this is indicative of us having to shift things around later. At some point we should support other surface geometry like spheres (which are supported by the VTK-m raycaster). The problem is that the structure of these outward facing classes do a bad job of mixing mapper types, which is what you really want here. I think there is a way around that, but we may eschew them altogether and interface directly with the raycast classes.